### PR TITLE
Reducer feature

### PR DESCRIPTION
**Pull request by ulisesedu7**
In this project, I've implemented Redux to add reducer functions and their respective actions for adding, removing, and updating books. Also, I've added reducers for the categories page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
-# Math-Magicians
+# BookStore
 
 This repository contains an web app for a books list to help you adding list of books and checking their progress so far.
```
Comment (Gambit142) [inline README.md]: - [x] Kindly add instructions for installing all the project's dependencies and starting the development server after the **_Getting Started Section_** of your `README.md` file. 👍

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { configureStore, combineReducers } from '@reduxjs/toolkit';
```
Comment (Gambit142) [inline src/redux/configureStore.js]: - [x] `combineReducers` should be imported from `redux` and not `@reduxjs/toolkit`. Kindly make this change. 👍  

```javascript
import { configureStore } from '@reduxjs/toolkit'
import { combineReducers } from 'redux'
```

File: src/redux/configureStore.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+// Establish the root reducer
+const rootReducer = combineReducers({
+  booksReducer,
+  statusReducer,
+});
+
+// Set up the store
+const store = configureStore(rootReducer);
+
```
Comment (Gambit142) [inline src/redux/configureStore.js]: - [x] [Required] Configure store accepts a single configuration object that contains a reducer key and some other keys. To configure your store, you need to pass the `rootReducer` variable as a value to the reducer key when using the `configureStore.` 👍 

#### Hint: `const store = configureStore({reducer: rootReducer})`. Kindly make adjustments. 👍 

- [x] [Optional] `configureStore` automatically calls `combineReducers` for you, so you might not want to do that directly. Your code can look like this:

```javascript
const store = configureStore({
  reducer: {
   books: booksReducer,
   categories: statusReducer
  },
});
```

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @ulisesedu7 👋, 

Good job so far!

## Highlights
- `Redux toolkit` is present in `package.json` ✔️ 
- The structure of the files follows **_duck pattern_** ✔️
- Created `Action types`, `Action creators` and `reducers` correctly ✔️   

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider the optional suggestions I made. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by ulisesedu7** [comment]
@Gambit142 Thank you very much for pointing all the changes, I'll start working on those to fix them!

File: src/redux/configureStore.js (hunk 3 above)
Comment (ulisesedu7) [reply src/redux/configureStore.js]: @Gambit142 Thank you very much for the hint!

**Comment by Gambit142** [comment]
> @Gambit142 Thank you very much for pointing all the changes, I'll start working on those to fix them!

Hi @ulisesedu7, I am glad I could be of help.

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @ulisesedu7  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ulisesedu7** [comment]
@zainsadaqat Super thanks!! Have a nice day!

**Pull request merged**